presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -89,6 +92,18 @@ public RuleAssert withSession(Session session)
         return this;
     }
 
+    public RuleAssert withStats(String nodeId, PlanNodeStatsEstimate nodeStats)
+    {
+        ((TestingStatsCalculator) statsCalculator).stats.put(new PlanNodeId(nodeId), nodeStats);
+        return this;
+    }
+
+    public RuleAssert withStatsCalculator(StatsCalculator statsCalculator)
+    {
+        this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
```

Comment:
this becomes tricky depending on order of how `RuleAssert` methods are invoked. Maybe `RuleAssert` should collect `stats` on it is own and `TestingStatsCalculator` should be created in `matches` method.

Reply:
Good catch. I commented on this in https://github.com/prestodb/presto/pull/10322#discussion_r179693561